### DISCUSSION TITLE
Apply base classes only to a certain scope

### DISCUSSION THREAD

**Discussion by JosepBernad** [tailwindlabs/tailwindcss#11208] [Help]
Hello everyone!

First of all, I would like to express my gratitude to the entire team for developing such an amazing tool! It has truly made implementing beautiful designs much more efficient and enjoyable.

I am currently developing an SDK that includes some UI components built with Tailwind UI. This SDK is designed to overlay on any website, providing additional functionality and features. However, I have encountered an issue that I hope you can help me with.

The problem I am facing is related to scoping the Tailwind base classes so that they only apply to specific elements within the HTML. I want to ensure that the integration of the SDK does not interfere with the existing styles of the website it is being implemented on.

Here's what I've tried so far:

1. I attempted to remove the base class from the generated CSS and apply it manually to the elements within my SDK. However, this solution proved to be suboptimal and not easily maintainable.

2. I started using the `prefix` configuration but it does not apply to the base classes.

I would greatly appreciate any guidance or suggestions on how to tackle this issue more effectively. Ideally.

Thank you so much,
Josep Bernad 🤘

**Comment by domdomegg** [comment]
Related: https://github.com/tailwindlabs/tailwindcss/discussions/7050, https://github.com/tailwindlabs/tailwindcss/discussions/7436

**Comment by itse4elhaam** [comment]
Try this


https://www.npmjs.com/package/tailwindcss-scoped-preflight
